### ISSUE TITLE
"--" not accepted as a comment

### ISSUE THREAD

**Issue by mdeweerd** [sqlfluff/sqlfluff#5724]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following contents works fine for mysql (`mysql TABLE < test.sql`) but sqlfluff fails on an "empty" comment line with 

```
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
```

Test case:
```sql
-- Test comment
--
-- Further comment


CREATE TABLE abcd(
        rowid integer AUTO_INCREMENT PRIMARY KEY NOT NULL,
        tms timestamp DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
) ENGINE=innodb;
```

Configuration pyproject.toml
```
[tool.sqlfluff.core]
ignore_comment_lines = true
sql_file_exts = ".sql"
verbose = 1
exclude_rules = "LT01,CP01,RF04"
dialect = "mysql"

[tool.sqlfluff.indentation]
indent_unit = "tab"
```

### Expected Behaviour

Should skip "empty" comments

### Observed Behaviour

Reports:

WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?

### How to reproduce

Test case:
```sql
-- Test comment
--
-- Further comment


CREATE TABLE abcd(
        rowid integer AUTO_INCREMENT PRIMARY KEY NOT NULL,
        tms timestamp DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
) ENGINE=innodb;
```

Configuration pyproject.toml
```
[tool.sqlfluff.core]
ignore_comment_lines = true
sql_file_exts = ".sql"
verbose = 1
exclude_rules = "LT01,CP01,RF04"
dialect = "mysql"

[tool.sqlfluff.indentation]
indent_unit = "tab"
```

### Dialect

mysql

### Version

$ sqlfluff --version
sqlfluff, version 3.0.3
Python 3.9.16

### Configuration

```
[tool.sqlfluff.core]
ignore_comment_lines = true
sql_file_exts = ".sql"
verbose = 1
exclude_rules = "LT01,CP01,RF04"
dialect = "mysql"

[tool.sqlfluff.indentation]
indent_unit = "tab"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mdeweerd** [comment]
As it may be related, I add that the following reports "Unable to lex characters: 'çao'.  Add a space after "--" 'fixes' the notice:

```
-- Test comment
--Add, Curaçao 
```

**Comment by mdeweerd** [comment]
I just found a reference that indicates that standard SQL requires a space after '--' for comments, but comments without a space should be accepted under the mysql dialect - or there should be an option to allow it.

https://dev.mysql.com/doc/refman/8.0/en/ansi-diff-comments.html
